subsys/usb/class/hid/core.c

```diff
@@ -205,11 +224,28 @@ static int hid_on_set_report(struct usb_setup_packet *setup, s32_t *len,
 static int hid_on_set_protocol(struct usb_setup_packet *setup, s32_t *len,
 			       u8_t **data)
 {
-	LOG_DBG("Set Protocol callback");
+#ifdef CONFIG_USB_HID_BOOT_PROTOCOL
```

Comment:
This will never build by sanitiycheck as it has no test case in samples/subsys/usb/hid/src/sample.yaml nor the configuration is enabled in prj.conf. I think add `CONFIG_USB_HID_BOOT_PROTOCOL=y` to prj.conf is better solution, @carles, @finikorg ?

Reply:
It was implemented as an option, like `CONFIG_USB_DEVICE_SOF` (which is not supported by all platforms). I think both should be added to a test case or `CONFIG_USB_DEVICE_SOF` to test and `CONFIG_USB_HID_BOOT_PROTOCOL` to prj.conf (it is platform agnostic).

Reply:
@masz-nordic I think addition test case would increase runtime of the sanitycheck,  let's add CONFIG_USB_HID_BOOT_PROTOCOL to prj.conf. @finikorg Any objections?

Reply:
@jfischer-phytec-iot OK for me to add to prj.conf

Reply:
Added to prj.conf